src/gc-alloc-profiler.cpp

```diff
@@ -209,52 +72,8 @@ string _type_as_string(jl_datatype_t *type) {
 
 // === stack stuff ===
 
-vector<StackFrame> get_julia_frames(jl_bt_element_t *bt_entry) {
-    vector<StackFrame> ret;
-
-    size_t ip = jl_bt_entry_header(bt_entry);
-    jl_value_t *code = jl_bt_entry_jlvalue(bt_entry, 0);
-    if (jl_is_method_instance(code)) {
-        // When interpreting a method instance, need to unwrap to find the code info
-        code = ((jl_method_instance_t*)code)->uninferred;
-    }
-    if (jl_is_code_info(code)) {
-        jl_code_info_t *src = (jl_code_info_t*)code;
-        // See also the debug info handling in codegen.cpp.
-        // NB: debuginfoloc is 1-based!
-        intptr_t debuginfoloc = ((int32_t*)jl_array_data(src->codelocs))[ip];
-        while (debuginfoloc != 0) {
-            jl_line_info_node_t *locinfo = (jl_line_info_node_t*)
-                jl_array_ptr_ref(src->linetable, debuginfoloc - 1);
-            assert(jl_typeis(locinfo, jl_lineinfonode_type));
-            const char *func_name = "Unknown";
-            jl_value_t *method = locinfo->method;
-            if (jl_is_method_instance(method))
-                method = ((jl_method_instance_t*)method)->def.value;
-            if (jl_is_method(method))
-                method = (jl_value_t*)((jl_method_t*)method)->name;
-            if (jl_is_symbol(method))
-                func_name = jl_symbol_name((jl_sym_t*)method);
-
-            ret.push_back(StackFrame{
-                func_name,
-                jl_symbol_name(locinfo->file),
-                locinfo->line,
-            });
-
-            debuginfoloc = locinfo->inlined_at;
-        }
-    }
-    else {
-        // If we're using this function something bad has already happened;
-        // be a bit defensive to avoid crashing while reporting the crash.
-        jl_safe_printf("No code info - unknown interpreter state!\n");
-    }
-    return ret;
-}
-
 RawBacktrace get_raw_backtrace() {
-    static jl_bt_element_t bt_data[JL_MAX_BT_SIZE];
+    jl_bt_element_t *bt_data = (jl_bt_element_t*) malloc(JL_MAX_BT_SIZE);
 
     // TODO: tune the number of frames that are skipped
     size_t bt_size = rec_backtrace(bt_data, JL_MAX_BT_SIZE, 1);
```

Comment:
Though it would probably be nice to not be allocating such a large backtrace every time when we only need a small one.. Maybe we could still record the backtrace onto a single static large one, and then malloc and copy only the `bt_size` elements?

Maybe something like this?:

```suggestion
    static jl_bt_element_t bt_data_buffer[JL_MAX_BT_SIZE];

    // TODO: tune the number of frames that are skipped
    size_t bt_size = rec_backtrace(bt_data_buffer, JL_MAX_BT_SIZE, 1);
    
    jl_bt_element_t *bt_data = (jl_bt_element_t*) malloc(bt_size * sizeof(jl_bt_element_t));

    std::copy(bt_data_buffer, bt_data_buffer + bt_size, bt_data);
```
Something like that?

Reply:
This way you're only allocating ~800 bytes (assuming ~100 frames) instead of 80,000 bytes?